newIDE/app/src/UI/SemiControlledAutoComplete.js

```diff
@@ -137,221 +65,192 @@ type State = {|
   inputValue: string | null,
 |};
 
-/**
- * An autocomplete field, showing options as the user type (or when the user presses down button).
- * Options can be chosen with keyboard or mouse.
- *
- * Supports divider between items and special items with `onClick` prop that when present is called
- * when the item is selected (and value is not changed).
- */
 export default class SemiControlledAutoComplete extends React.Component<
   Props,
   State
 > {
   _input = React.createRef<HTMLInputElement>();
+
   state = {
     inputValue: null,
   };
 
-  /**
-   * Allow to override the value being written in the autocomplete, if any.
-   * Usually you don't want to do that as the point of having a "SemiControlled" auto-complete
-   * is that the value being written by the user is the source of truth.
-   *
-   * In some cases, you want to force a new value when the auto-complete is focused (for example,
-   * if you opened a native dialog selector that is going to make the auto-complete focused/blurred/
-   * focused again, in a manner that depends on the OS).
-   * Call this function to do so.
-   *
-   * @param newValue The new value to set in the auto complete. If the field is not being focused,
-   * nothing will be forced (the value props will be used when the field is focused).
-   */
-  forceInputValueTo(newValue: string) {
+  focus() {
+    if (this._input.current) {
+      this._input.current.focus();
+      this._input.current.onkeydown = e => {
```

Comment:
Never ever add manually an event handler. This can create a memory leak or weird issues. Always use React event handlers, by passing a `onKeyDown` prop to the input

Reply:
So, should I add it to the inputProps instead of the _input ref?

Reply:
I think so. All events must be attached using props. Ref is just an escape hatch to "imperatively" say something to do the component (in most cases, it's for telling the component to focus, like here).